R/Figures.R

```diff
@@ -1127,6 +1127,340 @@ TADA_TwoCharacteristicScatterplot <- function(.data, id_cols = "TADA.ComparableD
         ), "<br>"
       )
     )
+  
+  return(scatterplot)
+}
+  
+#' Create A Scatterplot of Multiple Field Groups within a Column
+#'
+#' @param .data TADA data frame containing the data downloaded from the WQP,
+#'   where each row represents a unique data record. Data frame must include the
+#'   columns 'TADA.ComparableDataIdentifier', 'TADA.ResultMeasureValue', and 'TADA.ResultMeasure.MeasureUnitCode'
+#'   to run this function.
+#'
+#' @param id_cols The column in the dataset used to identify the unique groups to
+#'   be plotted. Defaults to 'TADA.ComparableDataIdentifier'. This input is flexible,
+#'   however, for the specific use case in the TADAShiny app where a user might
+#'   create groups based on a concatenation of the comparable data identifier
+#'   with other additional grouping variables (e.g. site type, site name, year,
+#'   organization, etc.)
+#'
+#' @param groups A vector of two identifiers from the id_cols column. For
+#'   example, if the id_cols is 'TADA.ComparableDataIdentifier', the groups could
+#'   be 'DISSOLVED OXYGEN (DO)_NA_NA_UG/L' and 'PH_NA_NA_NA'. These groups will
+#'   be specific to your dataset. If the id_cols is 'MonitoringLocationName',
+#'   the groups could be 'Upper Red Lake: West' and 'Upper Red Lake: West-Central'.
+#'
+#' @return A single plotly scatterplot figure with one x-axis (Date/Time) and a
+#'   left axis showing the units of a single characteristic groups
+#'   plotted on the same figure area wit a legend for each groups
+#'
+#' @export
+#'
+#' @examples
+#' # Load example dataset:
+#' data(Data_Nutrients_UT)
+#' # Create a single scatterplot with two monitoring locations (groups) for a single TADA.ComparableDataIdentifier. 
+#' # Dataframe has multiple comparable data identifier that was not filtered, select value for 'NITROGEN_TOTAL_AS N_MG/L' to plot this characteristic value.
```

Comment:
Example needs to show how to "select value for 'NITROGEN_TOTAL_AS_N_MG/L" as now that the documentation for this function is being created, this example is failing to run.

I was not sure how to select the value because Monitoring Locations were already specified in the groups argument.

![image](https://github.com/user-attachments/assets/caba030a-b6ec-48b3-9368-0520d547d38b)

Reply:
Could it be made to run more like TADA_Boxplot where if there is more than one TADA.ComparableDataIdentifier, a list of plots is created?

This is from the Create Boxplot(s) documentation: "The user can include
#'   additional grouping columns in the id_cols input. If more than one group
#'   exists in the data frame (i.e. two or more unique comparable data
#'   identifiers), the function creates a list of plots, where each list element
#'   name is a unique group identifier."

Reply:
You are definitely on the right track with wanting to give users the ability to choose a single one to plot or to generate multiple plots.

I think generating the list of plot as in TADA_Boxplot, TADA_Scatterplot, and TADA_Histogram is the way to go here too. That will keep things consistent within the basic package figure functions and lend itself more easily to incorporation into a automated work flow, rather than requiring the manual selection of a TADA.ComparableDataIdentifier. The examples from these functions highlight that using a data frame filtered for one TADA.ComparableDataIdentifier will result in one plot, while using an unfiltered data frame will result in a list of plots.